DNN Platform/Library/Services/Search/Internals/LuceneControllerImpl.cs

```diff
@@ -274,40 +277,61 @@ public LuceneResults Search(LuceneSearchContext searchContext)
             var maxResults = searchContext.LuceneQuery.PageIndex * searchContext.LuceneQuery.PageSize;
             var minResults = maxResults - searchContext.LuceneQuery.PageSize + 1;
 
-            var searcher = GetSearcher();
-            var searchSecurityTrimmer = new SearchSecurityTrimmer(new SearchSecurityTrimmerContext
-                {
-                    Searcher = searcher,
-                    SecurityChecker = searchContext.SecurityCheckerDelegate,
-                    LuceneQuery = searchContext.LuceneQuery,
-                    SearchQuery = searchContext.SearchQuery
-                });
-            searcher.Search(searchContext.LuceneQuery.Query, null, searchSecurityTrimmer);
-            luceneResults.TotalHits = searchSecurityTrimmer.TotalHits;
-
-            if (Logger.IsDebugEnabled)
+            for (var i = 0; i < _searchRetryTimes; i++)
             {
-                var sb = GetSearcResultExplanation(searchContext.LuceneQuery, searchSecurityTrimmer.ScoreDocs, searcher);
-		        Logger.Trace(sb);
-            }
+                try
+                {
+                    var searcher = GetSearcher();
+                    var searchSecurityTrimmer = new SearchSecurityTrimmer(new SearchSecurityTrimmerContext
+                    {
+                        Searcher = searcher,
+                        SecurityChecker = searchContext.SecurityCheckerDelegate,
+                        LuceneQuery = searchContext.LuceneQuery,
+                        SearchQuery = searchContext.SearchQuery
+                    });
+                    searcher.Search(searchContext.LuceneQuery.Query, null, searchSecurityTrimmer);
+                    luceneResults.TotalHits = searchSecurityTrimmer.TotalHits;
+
+                    if (Logger.IsDebugEnabled)
+                    {
+                        var sb = GetSearcResultExplanation(searchContext.LuceneQuery, searchSecurityTrimmer.ScoreDocs, searcher);
+                        Logger.Trace(sb);
+                    }
 
-            //Page doesn't exist
-            if (luceneResults.TotalHits < minResults)
-                return luceneResults;
```

Comment:
Why increase the nesting depth with this change from a return if the total hits were less than mins?

Reply:
this is not a new change in this PR.

Reply:
@zyhfish Yes, this is a change in this PR.  You removed the old process which returned early, to an inversed logic that increases the IL nesting depth of the code

Reply:
It's amusing that with the logical expression reversed the comment above is not incorrect :)